src/Storages/ObjectStorage/DataLakes/Iceberg/ManifestFile.h

```diff
@@ -43,17 +40,41 @@ struct ColumnInfo
     std::optional<DB::Range> hyperrectangle;
 };
 
-using FileEntry = std::variant<DataFileEntry>; // In the future we will add PositionalDeleteFileEntry and EqualityDeleteFileEntry here
+struct DataFileSpecificInfo
+{
+};
+
+struct PositionalDeleteFileSpecificInfo
+{
+    std::optional<String> reference_file_path;
+};
+
+using IcebergFileSpecificInfo = std::variant<DataFileSpecificInfo, PositionalDeleteFileSpecificInfo>;
```

Comment:
It doesn't make sense to delare an empty type `DataFileSpecificInfo`. Can we just declare a type like
```
struct DeletesFileSpecificInfo
{
    std::optional<String> reference_file_path; /// for postional deletes
    std::vector<Int32> equality_ids; /// for equality deletes
}
```